include/boost/url/pct_string_view.hpp

```diff
@@ -173,6 +172,36 @@ class pct_string_view final
     {
     }
 
+    /** Constructor
+
+        The newly constructed string will
+        reference the specified character
+        buffer. Ownership is not transferred.
+
+        @par Postconditions
+        @code
+        this->data() == s
+        @endcode
+
+        @par Complexity
+        Linear in `s.size()`.
+
+        @par Exception Safety
+        Exceptions thrown on invalid input.
+
+        @throw system_error
+        The string contains an invalid percent encoding.
+
+        @param s The string to construct from.
+    */
+    template <class Allocator>
+    pct_string_view(
+        std::basic_string<char, Allocator> const& s)
```

Comment:
something ain't right here... if you are going to add a templated constructor then shouldn't it be anything which is convertible to `urls::string_view`?

Reply:
> shouldn't it be anything which is convertible to urls::string_view

That became ambiguous now. For the same reason, we need the `pct_string_view(char const* s)` just above this one.

Reply:
why not
```
template<
    class StringLike
    , class = typename std::enable_if<
        std::is_convertible< StringLike, string_view >::value>:: type
>
pct_string_view( StringLike const& );
```